### ISSUE TITLE
[BUG] Remove workarounds from GH Actions

### ISSUE THREAD

**Issue by lechnerc77** [SAP-samples/btp-setup-automator#411]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest docker image for BTPSA?

- [X] I'm using the latest docker image for BTPSA.

### Which area is mainly impacted

Docker Image (Build or Run)

### Current Behavior

Due to several issues in dependencies of the Docker image build we had to introduce several workarounds. The workarounds comprise two areas:

(1) GH Action workflow definitions that use `docker/setup-buildx-action@v2`. Here we added the options:

```yaml
 with:
          driver-opts: |
            image=moby/buildkit:v0.10.6
``` 
This is necessary to avoid error when pushing the Docker images to the GitHub packages

(2) The `Dokcerfile` install the npm package `@sap/cds-dk`. With version 6.4.1. the package breaks the build of the Docker image for the ARM64 architecture due to a HANA client dependency. 
As workaround we pinned the version to **6.4.0** which allows the build of the Docker image for the ARM64 architecture

### Expected Behavior

The workarounds in the workflow definitions as well as in the Dockerfile are removed, namely the setup option in (1) and the fixed version of `@sap/cds-dk` of (2) are removed.

### Steps To Reproduce

n.a.

### Logs and configuration files available?

n.a.

### Anything else?

n.a.

**Comment by lechnerc77** [comment]
Partially removable with cds-dk 6.5.0 that uses [hdi-deploy 4.5.3](https://www.npmjs.com/package/@sap/hdi-deploy/v/4.5.3)

**Comment by lechnerc77** [comment]
"fixed" with PR #419 : ARM64 build is removed
